### PR TITLE
#9 user msvc core functionality 

### LINKED ISSUE #9: User Microservice MVSC

### PR DESCRIPTION
**Pull request by ZhiLinR**
file and branch renamed - unclear between actual scenario generator and just writing to profile 😔
controller, model and routes for registration created

### CODE REVIEW

**Review by ZhiLinR — COMMENTED** [review]
i forgot to include `commit:2dcca1a - separated profile and user` contains update profile functionality

**Comment by cre8tion** [comment]
> i forgot to include `commit:2dcca1a - separated profile and user` contains update profile functionality

Yeah its fine

File: microsvcs/user/app/controller/profile_con.js [hunk 1]
```diff
@@ -0,0 +1,42 @@
+// Import Database Functions
+const QUERIES = require("../model/profile.js")
+
+// FUNCTION EXPORTS --------------------------------------------------------
```
Comment (cre8tion) [inline microsvcs/user/app/controller/profile_con.js]: Feel free to remove comments about simple functions or code

File: microsvcs/user/app/controller/user_con.js [hunk 2]
```diff
@@ -0,0 +1,32 @@
+// Import Database Functions
+const QUERIES = require("../model/user.js")
+
+// LOCAL FUNCTIONS --------------------------------------------------------
```
Comment (cre8tion) [inline microsvcs/user/app/controller/user_con.js]: Can remove such comments too

File: microsvcs/user/app/routes.js [hunk 3]
```diff
@@ -0,0 +1,16 @@
+// routes.js
+const DEF_ROUTE = process.env.API_PATH.concat(process.env.ROUTE);
+
+module.exports = app => {
+    const USER = require("./controller/user_con.js");
+    const PROFILE = require("./controller/profile_con.js");
+    var router = require("express").Router();
+
+    router.get("/getProfile/:uid", PROFILE.getProfileByUID);
```
Comment (cre8tion) [inline microsvcs/user/app/routes.js]: /profile is fine (Try to keep to nouns basically)

You can read up more on REST APIs, but you can just reuse same name for GET, POST, PUT, DELETE etc.

For example:

GET /profile for retrieving your own profile details
PUT /profile for modifying your own profile details

GET /users/{userid} for accessing a specific user
POST /user for creating a new user etc.

Basically, try to standardise them. There are no correct answers haha, including mine

You can refer to this URL to learn more as well:
https://restfulapi.net/resource-naming/

**Comment by cre8tion** [comment]
I will make some basic changes to issues and this MR which i hope you can reference in the future.

File: microsvcs/user/README.md [hunk 4]
```diff
@@ -0,0 +1,27 @@
+# quick reference about API calls
+
+## GET api/user/getProfile/:uid
```
Comment (cre8tion) [inline microsvcs/user/README.md]: Dont need to worry about writing all this in the README for now, can just remain as it is.

**Comment by cre8tion** [comment]
Just try to clean up the rest endpoints naming a bit and should be fine.

**Review by cre8tion — APPROVED** [review]
LGTM, Feel free to merge in on your own

**Comment by ZhiLinR** [comment]
epic thank

**Pull request merged**